lib/createWorkers.js

```diff
@@ -47,22 +49,81 @@ function createWorkers(handlers, config, options = {}) {
 
   let workerConnection;
   const workerChannels = [];
+
+  const TASK_COMPLETED = 'task.completed';
+  const TASK_RETRIED = 'task.retried';
+  const TASK_FAILED = 'task.failed';
+  const WORKER_CLOSED = 'task.closed';
+
   const logger = configuration.logger;
 
   return {
     listen: co.wrap(_listen),
-    close: co.wrap(_close)
+    close: co.wrap(_close),
+    wait: _wait,
+    TASK_COMPLETED,
+    TASK_RETRIED,
+    TASK_FAILED,
+    WORKER_CLOSED
   };
 
   /**
+   * Wait for an event for a given amount of time.
+   *
+   * @param {string} eventName The name of the event to wait for.
+   * @param {number} timeout The maximum number of milliseconds to wait for, defaults to 1000.
+   * @return {Promise} A promise which is resolved when the event emitted, or rejected
+   * if the timeout occurs first.
+   * @private
+   */
+  function _wait(eventName, timeout = 1000) {
+    return new Promise((resolve, reject) => {
+      let timeoutId = null;
+
+      timeoutId = setTimeout(() => {
+        timeoutId = null;
+        reject(new Error(`event ${eventName} didn't occur after ${timeout}ms`));
+      }, timeout);
+
+      emitter.once(eventName, () => {
+        if (timeoutId !== null) {
+          clearTimeout(timeoutId);
+          resolve();
+        }
+      });
+    });
+  }
+
+  /**
+   * Emit an event with the provided name.
+   *
+   * Emission is scheduled with process.nextTick to allow current work to complete.
+   *
+   * @param {string} eventName the name of the event to publish
+   * @returns {void}
+   */
+  function _emit(eventName) {
+    process.nextTick(() => emitter.emit(eventName));
+  }
+
+  /**
+   * Exits the process after having emited the worker close event.
+   * @returns {void}
+   */
+  function _exit() {
+    _emit('worker.closed');
```

Comment:
Due to the `process.nextTick` and the `process.exit(0)` the line after, this event will never be catched.

Reply:
I would use the `setTimeout` in the `_exit` method:

```js
setTimeout(() => process.exit(0), configuration.processExitTimeout);
```

Reply:
ok, nice catch